Map/MapManager.gd

```diff
@@ -0,0 +1,28 @@
+extends Node2D
+## Class to manage backend for rooms (generation and such)
+ 
+#layout changes the width of the map's floors
+static func create_map(_layout: Array[int]) -> MapBase: ## Generates and Populates a map with rooms that have random room types. More in depth algorithms will be added in the future
+	var _map: MapBase = MapBase.new()
+	var _grid: Array[Array] = [] ## 2d array to return. this will be populated with rooms
+	var _max_floor_size: int = _layout.max()
+	for index_height: int in range(_layout.size()): ## loop through the floor and append null to create a square grid
+		_grid.append([])
+		
+		var _floor_width : int = _layout[index_height]
+		var _padding_size : int = (_max_floor_size - _floor_width)/2
+		
+		_grid[index_height].resize(_max_floor_size-_floor_width)
+		_grid[index_height].fill(null)
+		
+		for index_width: int in range(_layout[index_height]):## loop through elements of the grid and assign a room type
+			var _rand_type_index: int = randi() % GlobalVar.EVENTS_CLASSIFICATION.size()
```

Comment:
Would `randi_range(0, GlobalVar.EVENTS_CLASSIFICATION.size() - 1)` be better here? `randi()` can return `2^32 - 1` so not sure if we want to be doing that big of a calculation for every room we go through.
`

Reply:
you are right, a range is probably better